File: app/src/main/java/com/example/github_api_concrete/view/activities/SplashActivity.java
package com.example.github_api_concrete.view.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageView;

import com.example.github_api_concrete.R;

import java.util.Timer;
import java.util.TimerTask;

public class SplashActivity extends AppCompatActivity {

    private Timer timer = new Timer();
    private ImageView splash;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        splash = findViewById(R.id.imageSplash);

        splash.setOnClickListener(v -> {
            jump();
        });

        timer.schedule(new TimerTask() {
            @Override
            public void run() {
                jump();
            }
        },3000);
    }

    private void jump(){
        timer.cancel();
        startActivity(new Intent(SplashActivity.this, ReposActivity.class));
        finish();
    }
}

File: githubtest/app/src/main/java/com/example/githubtest/features/splash/SplashActivity.kt
package com.example.githubtest.features.splash

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.animation.AnimationUtils
import android.widget.ImageView
import com.example.githubtest.R
import com.example.githubtest.features.repository.RepositoryActivity
import java.util.*

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        Timer().schedule(object : TimerTask() {
            override fun run() {
                jump(null)
            }
        }, 5000)
        val image = findViewById(R.id.imageSplash) as ImageView
        val animation = AnimationUtils.loadAnimation(
            applicationContext,
            R.anim.animation_blink
        )
        image.startAnimation(animation)
    }

    fun jump(view: View?) {
        startActivity(Intent(this, RepositoryActivity::class.java))
        finish()
    }
}

